proxy/src/auth/backend/console.rs

```diff
@@ -251,15 +255,36 @@ impl Api<'_> {
                 .dbname(self.creds.dbname)
                 .user(self.creds.user);
 
-            Ok(NodeInfo {
+            let node = NodeInfo {
                 config,
-                aux: body.aux,
-            })
+                aux: body.aux.into(),
+            };
+
+            Ok(node)
         }
         .map_err(crate::error::log_error)
         .instrument(info_span!("wake_compute", id = request_id))
         .await
     }
+
+    /// Wake up the compute node and return the corresponding connection info.
+    pub async fn wake_compute(&self) -> Result<CachedNodeInfo, WakeComputeError> {
+        let key = self.creds.project().expect("impossible");
+
+        // Every time we do a wakeup http request, the compute node will stay up
+        // for some time (~5 mins; depends on the console's scale-to-zero policy);
+        // The connection info remains the same during that period of time,
+        // which means that we might cache it to reduce the load and latency.
```

Comment:
> for some time (~5 mins; depends on the console's scale-to-zero policy)

I realized that now it's not strictly correct, or not correct anymore.

Some context, we use NodePort service (it exposes some fixed port on **every** k8s node) and ip address of some random not tainted k8s node as an entry point. But, we started enabling k8s autoscaler, so k8s nodes are not something really persistent, they can appear to handle increased load and disappear, when resources are not needed anymore.

That said, previously picked address of some k8s node may become invalid. To handle that, I added some ad-hoc service discovery into activity monitor in the console. Now on prod it's configured to poke every presumably running compute every ~10 secs, and if it doesn't respond, it tries to pick a different k8s node address and poke again.

Currently, k8s autoscaler is rolled out to `us-east-2` only, but soon to be in every region, as it seems to be decently working.
- The rate of such address updates could be find on this metric https://neonprod.grafana.net/d/S3MPUUonz/neon-console?orgId=1&viewPanel=57&from=now-24h&to=now
- And here is a rate of failed checks (even after address update) https://neonprod.grafana.net/d/S3MPUUonz/neon-console?orgId=1&viewPanel=55&from=now-24h&to=now

Since proxy has some force invalidation after failed connection, that's probably fine, I just stumbled on this comment and decided to clarify some things. Everything explained above can change one day (e.g. we can switch to using stable ClusterIP addresses), but that's how it works right now.

So the best flow could be
- accept client connection
- try connecting using cached address
- if not, try to do wake up without dropping the client connection
Probably that's how it's done already, then cool, otherwise it may increase error rate in client connection.

Reply:
Ok, thanks for the information.

> Since proxy has some force invalidation after failed connection, that's probably fine, I just stumbled on this comment and decided to clarify some things. Everything explained above can change one day (e.g. we can switch to using stable ClusterIP addresses), but that's how it works right now.

I'm starting to feel very uneasy here.

* This info arrived very late in the dev cycle. In your opinion, how do we fix that?
* What's the proposed default TTL of a cache entry, then? 10s?
* Say, we pick ttl=10s, how do we keep it up-to-date? Should we assign a maintainer from @neondatabase/control-plane?

Reply:
I have a proposal (which IMO has been posted elsewhere, but nonetheless):

What do you think about returning the proposed TTL from `proxy_wake_compute`? This would give us the flexibility we need. Each entry might have its own TTL.

Reply:
Unfortunately, I don't think that any TTL will guarantee that address will last this time for real. I once answered / briefly explained this in the Slack (somewhat around this thread https://neondb.slack.com/archives/C04JRCGV8F2/p1673357467738899?thread_ts=1673274880.738879&cid=C04JRCGV8F2)

There are certain user actions + undergoing background activities in the k8s, which can invalidate it almost immediately after proxy cached it. So 5 mins default TTL for a happy path sounds good to me, but having an address refresh backup path seems to be a must have

Reply:
Yeah, this makes sense. I'm going to give this another spin using the new intel.

Reply:
Done.